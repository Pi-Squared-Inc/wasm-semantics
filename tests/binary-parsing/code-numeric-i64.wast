(module $mymodule
  (func (param i32) (result i32)
    i64.const 0
    drop

    i64.const 0
    i64.clz
    drop

    i64.const 0
    i64.ctz
    drop

    i64.const 0
    i64.popcnt
    drop

    i64.const 1
    i64.const 1
    i64.add
    drop

    i64.const 1
    i64.const 1
    i64.sub
    drop

    i64.const 1
    i64.const 1
    i64.mul
    drop

    i64.const 1
    i64.const 1
    i64.div_s
    drop

    i64.const 1
    i64.const 1
    i64.div_u
    drop

    i64.const 1
    i64.const 1
    i64.rem_s
    drop

    i64.const 1
    i64.const 1
    i64.rem_u
    drop

    i64.const 1
    i64.const 1
    i64.and
    drop

    i64.const 1
    i64.const 1
    i64.or
    drop

    i64.const 1
    i64.const 1
    i64.xor
    drop

    i64.const 1
    i64.const 1
    i64.shl
    drop

    i64.const 1
    i64.const 1
    i64.shr_s
    drop

    i64.const 1
    i64.const 1
    i64.shr_u
    drop

    i64.const 1
    i64.const 1
    i64.rotl
    drop

    i64.const 1
    i64.const 1
    i64.rotr
    drop

    i64.const 1
    i64.eqz
    drop

    i64.const 1
    i64.const 1
    i64.eq
    drop

    i64.const 1
    i64.const 1
    i64.ne
    drop

    i64.const 1
    i64.const 1
    i64.lt_s
    drop

    i64.const 1
    i64.const 1
    i64.lt_u
    drop

    i64.const 1
    i64.const 1
    i64.gt_s
    drop

    i64.const 1
    i64.const 1
    i64.gt_u
    drop

    i64.const 1
    i64.const 1
    i64.le_s
    drop

    i64.const 1
    i64.const 1
    i64.le_u
    drop

    i64.const 1
    i64.const 1
    i64.ge_s
    drop

    i64.const 1
    i64.const 1
    i64.ge_u
    drop

    (return (i32.const 1))
  )
)
