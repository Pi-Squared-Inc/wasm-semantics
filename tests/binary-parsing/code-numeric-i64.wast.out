<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i64 . const 0  drop  i64 . const 0  i32 . clz  drop  i64 . const 0  i32 . ctz  drop  i64 . const 0  i32 . popcnt  drop  i64 . const 1  i64 . const 1  i64 . add  drop  i64 . const 1  i64 . const 1  i64 . sub  drop  i64 . const 1  i64 . const 1  i64 . mul  drop  i64 . const 1  i64 . const 1  i64 . div_s  drop  i64 . const 1  i64 . const 1  i64 . div_u  drop  i64 . const 1  i64 . const 1  i64 . rem_s  drop  i64 . const 1  i64 . const 1  i64 . rem_u  drop  i64 . const 1  i64 . const 1  i64 . and  drop  i64 . const 1  i64 . const 1  i64 . or  drop  i64 . const 1  i64 . const 1  i64 . xor  drop  i64 . const 1  i64 . const 1  i64 . shl  drop  i64 . const 1  i64 . const 1  i64 . shr_s  drop  i64 . const 1  i64 . const 1  i64 . shr_u  drop  i64 . const 1  i64 . const 1  i64 . rotl  drop  i64 . const 1  i64 . const 1  i64 . rotr  drop  i64 . const 1  i64 . eqz  drop  i64 . const 1  i64 . const 1  i64 . eq  drop  i64 . const 1  i64 . const 1  i64 . ne  drop  i64 . const 1  i64 . const 1  i64 . lt_s  drop  i64 . const 1  i64 . const 1  i64 . lt_u  drop  i64 . const 1  i64 . const 1  i64 . gt_s  drop  i64 . const 1  i64 . const 1  i64 . gt_u  drop  i64 . const 1  i64 . const 1  i64 . le_s  drop  i64 . const 1  i64 . const 1  i64 . le_u  drop  i64 . const 1  i64 . const 1  i64 . ge_s  drop  i64 . const 1  i64 . const 1  i64 . ge_u  drop  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: .EmptyStmts , mems: .EmptyStmts , globals: .EmptyStmts , elem: .EmptyStmts , data: .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
